lib/system/io.nim

```diff
@@ -278,16 +278,19 @@ elif defined(posix) and not defined(lwip) and not defined(nimscript):
   proc c_fcntl(fd: cint, cmd: cint): cint {.
     importc: "fcntl", header: "<fcntl.h>", varargs.}
 elif defined(windows):
-  const HANDLE_FLAG_INHERIT = culong 0x1
+  const HANDLE_FLAG_INHERIT = 1'i32
+
   proc getOsfhandle(fd: cint): int {.
     importc: "_get_osfhandle", header: "<io.h>".}
 
   type
     IoHandle = distinct pointer
       ## Windows' HANDLE type. Defined as an untyped pointer but is **not**
       ## one. Named like this to avoid collision with other `system` modules.
-  proc setHandleInformation(handle: IoHandle, mask, flags: culong): cint {.
-    importc: "SetHandleInformation", header: "<handleapi.h>".}
+
+  proc setHandleInformation(hObject: IoHandle, dwMask: int32, dwFlags: int32):
```

Comment:
this should be:

```nim
type WinDWORD = culong
type WinBOOL = cint
proc setHandleInformation(hObject: IoHandle, dwMask, dwFlags: DWORD): WinBOOL {.
    importc: "SetHandleInformation", header: "<handleapi.h>".}
```

or at least:
```nim
proc setHandleInformation(hObject: IoHandle, dwMask, dwFlags: culong): cint {.stdcall, dynlib: "kernel32", importc: "SetHandleInformation".}
```

to match API in https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-sethandleinformation
and https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types

This should still fix the referenced issues since it was IIRC due to `header: "<handleapi.h>"`

Reply:
I had followed winlean, even though I had code using different types, because I also believe that the current winlean does not match the types of WinAPI. Thanks for the comments full of references.